frontend/src/app/scratch/[slug]/claim/page.tsx

```diff
@@ -0,0 +1,32 @@
+"use client"
+
+import { useEffect, useRef } from "react"
+
+import { useRouter } from "next/navigation"
+
+import LoadingSkeleton from "@/app/scratch/[slug]/loading"
+import { post } from "@/lib/api/request"
+
+export default function Page({ params, searchParams }: {
+    params: { slug: string }
+    searchParams: { token: string }
+}) {
+    const router = useRouter()
+
+    // The POST request must happen on the client so
+    // that the Django session cookie is present.
+    const effectRan = useRef(false)
+    useEffect(() => {
+        if (!effectRan.current) {
+            post(`/scratch/${params.slug}/claim`, { token: searchParams.token })
+                .catch(err => console.error(err))
```

Comment:
Need to do proper error handling; if there's an error (e.g. the scratch is already claimed) then this page will appear to load forever

Reply:
That's what the `.finally` block handles!

Reply:
If the claim fails I think we should display the error rather than silently redirect